### PR TITLE
Cherry Pick: DetailsList isSelectedOnFocus

### PR DESCRIPTION
**Pull request by spmonahan**
Cherry picks a change to DetailsList from `master` that adds `isSelectedOnFocus`.

Original pull request: #23467

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">7.0 Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/BaseButton-mount_base.html">990</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/BaseButton-mount.html">1101</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Breadcrumb-mount_base.html">43680</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Breadcrumb-mount.html">43097</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Checkbox-mount_base.html">1748</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Checkbox-mount.html">1758</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/CheckboxBase-mount_base.html">1524</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/CheckboxBase-mount.html">1456</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/ChoiceGroup-mount_base.html">5448</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/ChoiceGroup-mount.html">5509</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/ComboBox-mount_base.html">974</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/ComboBox-mount.html">982</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/CommandBar-mount_base.html">8184</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/CommandBar-mount.html">8127</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/ContextualMenu-mount_base.html">14276</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/ContextualMenu-mount.html">14285</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/DefaultButton-mount_base.html">1217</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/DefaultButton-mount.html">1218</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/DetailsRow-mount_base.html">4119</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/DetailsRow-mount.html">4079</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/DetailsRowFast-mount_base.html">4100</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/DetailsRowFast-mount.html">4054</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/DetailsRowNoStyles-mount_base.html">3843</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/DetailsRowNoStyles-mount.html">3794</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Dialog-mount_base.html">2508</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Dialog-mount.html">2461</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/DocumentCardTitle-mount_base.html">1848</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/DocumentCardTitle-mount.html">1930</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Dropdown-mount_base.html">2782</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Dropdown-mount.html">2725</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/FocusTrapZone-mount_base.html">1782</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/FocusTrapZone-mount.html">1787</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/FocusZone-mount_base.html">1914</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/FocusZone-mount.html">1861</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/IconButton-mount_base.html">1944</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/IconButton-mount.html">1914</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Label-mount_base.html">333</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Label-mount.html">335</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Layer-mount_base.html">2145</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Layer-mount.html">2099</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Link-mount_base.html">490</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Link-mount.html">482</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/MenuButton-mount_base.html">1615</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/MenuButton-mount.html">1612</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/MessageBar-mount_base.html">2132</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/MessageBar-mount.html">2119</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Nav-mount_base.html">3922</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Nav-mount.html">3537</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/OverflowSet-mount_base.html">1481</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/OverflowSet-mount.html">1522</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Panel-mount_base.html">1588</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Panel-mount.html">1543</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Persona-mount_base.html">997</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Persona-mount.html">1052</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Pivot-mount_base.html">1604</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Pivot-mount.html">1560</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/PrimaryButton-mount_base.html">1360</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/PrimaryButton-mount.html">1368</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Rating-mount_base.html">8564</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Rating-mount.html">8637</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/SearchBox-mount_base.html">1390</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/SearchBox-mount.html">1362</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Shimmer-mount_base.html">2794</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Shimmer-mount.html">2790</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Slider-mount_base.html">1580</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Slider-mount.html">1609</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/SpinButton-mount_base.html">5301</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/SpinButton-mount.html">5462</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Spinner-mount_base.html">423</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Spinner-mount.html">427</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/SplitButton-mount_base.html">3441</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/SplitButton-mount.html">3413</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Stack-mount_base.html">524</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Stack-mount.html">530</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/StackWithIntrinsicChildren-mount_base.html">1842</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/StackWithIntrinsicChildren-mount.html">1917</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/StackWithTextChildren-mount_base.html">5344</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/StackWithTextChildren-mount.html">5364</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/SwatchColorPicker-mount_base.html">11563</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/SwatchColorPicker-mount.html">11311</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/TagPicker-mount_base.html">3127</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/TagPicker-mount.html">3037</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/TeachingBubble-mount_base.html">52896</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/TeachingBubble-mount.html">52398</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Text-mount_base.html">469</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Text-mount.html">492</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/TextField-mount_base.html">1528</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/TextField-mount.html">1556</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Toggle-mount_base.html">890</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/Toggle-mount.html">861</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/button-mount_base.html">118</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23548/perf-test/button-mount.html">127</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: change/@fluentui-react-1c69b2a4-8b4f-43fd-bfb9-02d2307b9b4b.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "add isSelectedOnFocus prop to DetailsList",
```
Comment (sopranopillow) [inline change/@fluentui-react-1c69b2a4-8b4f-43fd-bfb9-02d2307b9b4b.json:3]:
```suggestion
  "comment": "feat: add isSelectedOnFocus prop to DetailsList",
```

**Review by sopranopillow — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

**Pull request merged**